### ISSUE TITLE
New line causes issues in listview

### ISSUE THREAD

**Issue by meldarionqeusse** [JackGruber/joplin-plugin-note-overview#24]
I was not quite sure what title to give the issue hope I can explain the issue clearer with pictures. Before I start I want to thank you for implementing the listview feature which is something I can personally recommended to you and you went so much above and beyond which I imagined. 

Back to the issue at hand... So basically I was trying to achieve this. So I want the title of each note and then details section which is hidden as in image 1. Details can be opened to show to body of the note as shown in image 2.

Image 1
![image](https://user-images.githubusercontent.com/21668964/128895696-6fc1043c-0a19-4982-a6ff-7f3236d1a8e4.png)

Image 2
![image](https://user-images.githubusercontent.com/21668964/128895964-e25a41dc-bfc6-4fe3-9a78-a657a223311e.png)

I am using the code below to achieve this
```
<!-- note-overview-plugin
search: notebook:"test"
fields: title, body
sort: title DESC
listview:
  text: "{{title}} <details> {{body}} </details>"
  separator: <hr>
-->
```

So the first issue as you can see in Image 2 the heading in the first instance (for the note named test 4) is not recognized as a heading but the heading in the note name note 2 is recognized. The difference is that in the note named test 4 I do not have a new line before the heading while in note 2 I have a new line. 

Second issue if I have a newline at the end of a note the subsequent note is not recognized as a link. In image 3 as you can see Note 2 doe snot have the Joplin icon but is shown as [Note 2](:/873b2a5a2a3546e3b37595551429658c) instead. The only thing I changed between Image 3 and Image 1 is that I added a newline at the end of Note 3 hence adding the new line at the end of Note 3 cause the link to Note  2 in the note overview render to break. 

Image 3
![image](https://user-images.githubusercontent.com/21668964/128896170-24e9191c-f0fc-4687-98ff-bf9e65ad5e44.png)

Long story short I think there is a problem with how newlines in notes are tackled, I think you might need to add some magic to your code to handle new lines at the end and beginning of a note. 

Hope I have given a clear explanation of the issue and sorry for the long one. Thanks for your incredible plugin and for taking up suggestions and implementing them. What you have done is already more than enough and a dream come true for me.

**Comment by JackGruber** [comment]
There are several solutions for this. Simply insert a new line via `\n` at seperator or `text`.
```
  separator: "<hr>\n"
```
or
```
  text: "\n{{title}} <details> {{body}} </details>"
```

**Comment by meldarionqeusse** [comment]
This gets changed   `separator: "<hr>\n"` after the note is created to 

```  
separator: |
    <hr>
```

Same with the second one it gets changed to 
```
  text: |-2
    
     {{title}} <details> {{body}} </details>
```


Also although the solution proposed the solves the second problem : 
![image](https://user-images.githubusercontent.com/21668964/129860172-1f1eb97f-4de2-446c-a488-b35aa18df0ae.png)

but does not solve issue 1 where the heading of the note is not rendered as a heading Image 1 see note 4 vs note 2. In the actual note in Note 2 therea newline before the heading but in note 4 there isn't
![image](https://user-images.githubusercontent.com/21668964/129860657-6e2415fc-c721-4ab8-9106-a4bcc950bd0c.png)

**Comment by meldarionqeusse** [comment]
I have solved the issue thanks for pointing me in the right direction. 
```
<!-- note-overview-plugin
search: notebook:"test"
fields: title, body
sort: title DESC
listview:
  text: "{{title}} <details> \r\n {{body}} </details>"
  separator: "<hr>\r\n"
-->
```

However it still gets changed to but I am not complaining cause it still works

```
<!-- note-overview-plugin
search: notebook:"test"
fields: title, body
sort: title DESC
listview:
  text: "{{title}} <details> \r

    \ {{body}} </details>"
  separator: "<hr>\r\n"
-->
```

**Comment by JackGruber** [comment]
That's ok, the configuration is written back by the YAML parser.
